src/transport/UnauthenticatedSessionTable.h

```diff
@@ -278,24 +278,28 @@ class UnauthenticatedSessionTable
     CHIP_ERROR AllocEntry(UnauthenticatedSession::SessionRole sessionRole, NodeId ephemeralInitiatorNodeID,
                           const ReliableMessageProtocolConfig & config, UnauthenticatedSession *& entry)
     {
-        auto entryToUse = mEntries.CreateObject(sessionRole, ephemeralInitiatorNodeID, config, *this);
-        if (entryToUse != nullptr)
+        entry = mEntries.CreateObject(sessionRole, ephemeralInitiatorNodeID, config, *this);
```

Comment:
I think this code was trying to not change the value of the outparam on error....

Reply:
Do you think that was intentional? There seems to be no documentation/guarantee on this.

Reply:
I am unsure why we would want this kind of guarantees.

We seem to always have AllocEntry to start with nullptr:
- line 256 has direct nullptr and line 229 does a FindEntry and calls AllocEntry only if result is nullptr.

Reply:
I added a comment to say return will be null on error.
I believe that is correct behavior and existing code does use it like that (it starts with null anyway). Since this is private, the only usages were lines 256 and 229.

Reply:
It may also be that the code was trying to avoid the error from casting from EntryType to the UnauthenticatedSession. The usage of `auto` hid that entryToUse is `EntryType == detail::UnauthenticatedSessionPoolEntry<kMaxSessionCount>` where as entry is `UnauthenticatedSession`.

I had to add a cast because of this ... if you think the cast is ugly I can revert to the previous "separate type" code.

Reply:
Yes, I would rather not have the cast.